bentoml/_internal/frameworks/keras.py

```diff
@@ -132,29 +138,41 @@ def load(
 
     """  # noqa
 
-    model = model_store.get(tag)
-    if model.info.module not in (MODULE_NAME, __name__):
+    bentoml_model: Model = model_store.get(tag)
+    if bentoml_model.info.module not in (MODULE_NAME, __name__):
         raise BentoMLException(
             f"Model {tag} was saved with module {model.info.module}, failed loading with {MODULE_NAME}."
         )
 
     default_custom_objects = None
-    if model.info.options["custom_objects"]:
-        assert Path(model.path_of(_CUSTOM_OBJ_FNAME)).is_file()
-        with Path(model.path_of(_CUSTOM_OBJ_FNAME)).open("rb") as dcof:
+    if bentoml_model.info.options["custom_objects"]:
+        assert Path(bentoml_model.path_of(_CUSTOM_OBJ_FNAME)).is_file()
+        with Path(bentoml_model.path_of(_CUSTOM_OBJ_FNAME)).open("rb") as dcof:
             default_custom_objects = cloudpickle.load(dcof)
 
+    model_format = bentoml_model.info.context.get("model_format")
+    if model_format:
+        # ignore version=v1 now because all version should be v1
+        save_format, store_as_json, _ = model_format.split(":")
+    # backward compatibility
+    else:
+        save_format = "h5"
+        store_as_json = bentoml_model.info.options["store_as_json"]
+
     with get_session().as_default():
-        if model.info.options["store_as_json"]:
-            assert Path(model.path_of(_MODEL_JSON_FNAME)).is_file()
-            with Path(model.path_of(_MODEL_JSON_FNAME)).open("r") as jsonf:
+        if store_as_json:
+            assert Path(bentoml_model.path_of(_MODEL_JSON_FNAME)).is_file()
+            with Path(bentoml_model.path_of(_MODEL_JSON_FNAME)).open("r") as jsonf:
                 model_json = jsonf.read()
             model = keras.models.model_from_json(
                 model_json, custom_objects=default_custom_objects
             )
+            weight_fname = _MODEL_WEIGHT_FNAME_MAPPING[save_format]
+            model.load_weights(bentoml_model.path_of(weight_fname))
         else:
+            model_fname = _SAVED_MODEL_FNAME_MAPPING[save_format]
             model = keras.models.load_model(
```

Comment:
does `load_model` accept a `SavedModel` format?

Reply:
We are also planning to have a `import_tf_savedmodel` for tensorflow as well as `import_from_pt` for pytorch